graph/iterator/edges.go

```diff
@@ -0,0 +1,129 @@
+// Copyright ©2018 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package iterator
+
+import "gonum.org/v1/gonum/graph"
+
+// OrderedEdges implements the graph.Edges and graph.EdgeSlicer interfaces.
+// The iteration order of OrderedEdges is the order of edges passed to
+// NewEdgeIterator.
+type OrderedEdges struct {
+	idx   int
+	edges []graph.Edge
```

Comment:
There is currently no way to append new edges to `OrderedEdges`, and that's probably fine.

Reply:
This is intentional.

Reply:
Though it may be worth adding a `Swap(i, j int)` method here (and to the other ordered iterators) that bases `i` and `j` on `idx`. This would allow arbitrary user orderings if the user embeds the value in a struct with a `Less`method.